Samples/ios/SampleApp/SampleAppiOS/SampleMSALUtil.m

```diff
@@ -0,0 +1,217 @@
+{
+    MSALPublicClientApplication *application = [self clientApplication];
+    
+    NSError *error = nil;
+    MSALUser *currentUser = [self currentUser:&error];
+    if (!currentUser)
+    {
+        acquireTokenBlock(nil, error);
+        return;
+    }
+    
+    // Depending on how this user has been used with this application before it is possible for there to be multiple
+    // tokens of varying authorities for this user in the cache. Because we are trying to get a token specifically
+    // for graph in this sample it's best to specify the user's home authority to remove any possibility of there
+    // being any ambiquity in the cache lookup.
+    NSString *homeAuthority = [NSString stringWithFormat:@"https://login.microsoftonline.com/%@", currentUser.utid];
+    
+    
+    [application acquireTokenSilentForScopes:@[@"User.Read"]
+                                        user:currentUser
+                                   authority:homeAuthority
+                                forceRefresh:NO
+                               correlationId:nil
+                             completionBlock:^(MSALResult *result, NSError *error)
+    {
+        acquireTokenBlock(result.accessToken, error);
+    }];
+}
+
++ (void)acquireTokenInteractiveForCurrentUser:(void (^)(NSString *token, NSError *error))acquireTokenBlock
+{
+    MSALPublicClientApplication *application = [self clientApplication];
+    
+    NSError *error = nil;
+    MSALUser *currentUser = [self currentUser:&error];
+    if (!currentUser)
+    {
+        acquireTokenBlock(nil, error);
+        return;
+    }
+
+    [application acquireTokenForScopes:@[@"User.Read"]
+                                  user:currentUser
+                            uiBehavior:MSALForceConsent
+                  extraQueryParameters:nil
+                       completionBlock:^(MSALResult *result, NSError *error)
+     {
+         acquireTokenBlock(result.accessToken, error);
+     }];
+}
+
++ (void)signOut
+
+{
+    MSALPublicClientApplication *application = [self clientApplication];
+    MSALUser *currentUser = [self currentUser:nil];
+    [self cleanupLocalState];
+    
+    // Signing out a user requires removing this from MSAL and cleaning up any extra state that the application
+    // might be maintaining outside of MSAL for the user.
+    
+    // This remove call only removes the user's tokens for this client ID in the local keychain cache. It does
+    // not sign the user completely out of the device or remove tokens for the user for other client IDs. If
+    // you have multiple applications sharing a client ID this will make the user effectively "disappear" for
+    // those applications as well if you are using Keychain Cache Sharing (not currently available in MSAL
+    // build preview). We do not recommend shating a ClientID among multiple apps.
+    [application removeUser:currentUser error:nil];
```

Comment:
I'd rather not have any comments here on specific changes we may or may not make to the API behavior.